pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -625,6 +652,169 @@ func (pc *PodController) initializeSecondaryInterfaceStore() error {
 	return nil
 }
 
+// initializeSRIOVSecondaryInterfaceStore restores secondary interfaceStore for SR-IOV interfaces
+// when agent restarts. It will get the Pod info from the store of primary interfaces, and check
+// the NetworkStatus annotation of a Pod, then restore the SR-IOV interfaces based on the NetworkAttachmentDefinition
+// name and device's pci_address in the NetworkStatus.
+func (pc *PodController) initializeSRIOVSecondaryInterfaceStore() error {
+	if ok, err := isNADCRDAvailable(pc.kubeClient); !ok || err != nil {
+		klog.ErrorS(err, "NetworkAttachmentDefinition CRD is not available")
+		return nil
+	}
+	allNADs, err := getAllNADs(pc.netAttachDefClient)
+	if err != nil {
```

Comment:
could we check the error instead of making a separate call? If an API doesn't exist, the error should be NotFound 404.

Reply:
Updated.